python_modules/libraries/dagster-dg/dagster_dg/context.py

```diff
@@ -136,33 +138,41 @@ def from_file_discovery_and_command_line_config(
         path: Path,
         command_line_config: DgRawCliConfig,
     ) -> Self:
-        config_path = discover_config_file(path)
+        root_config_path = discover_config_file(path)
         workspace_config_path = discover_config_file(
             path, lambda x: bool(x.get("directory_type") == "workspace")
         )
 
-        if config_path:
-            root_path = config_path.parent
-            root_file_config = load_dg_root_file_config(config_path)
-            if workspace_config_path:
-                workspace_root_path = workspace_config_path.parent
-
-                # If the workspace config is different from the root config, we need to load the
-                # workspace config.
-                container_workspace_file_config = (
-                    load_dg_workspace_file_config(workspace_config_path)
-                    if config_path != workspace_config_path
-                    else None
-                )
+        if root_config_path:
+            root_path = root_config_path.parent
+            root_file_config = load_dg_root_file_config(root_config_path)
```

Comment:
this was there before but is it always true that the workspace path is a direct parent of the project path? I thought we were imagining more complicated schemes where the projects were at odd locations and there were links in the workspace pyproject.toml to them - do we need to continue looking upwards?

Reply:
There is no assumption here that the workspace path is a direct parent, only that it's an ancestor-- so this is compatible with the scheme in https://github.com/dagster-io/dagster/pull/27477 where the workspace `pyproject.toml` links to projects.

It isn't compatible with situations where the project is completely outside of the workspace, that's something we'll need to think about (if we want to support it at all).